includes/standard-source.php

```diff
@@ -0,0 +1,186 @@
+	 * @return string
+	 */
+	public static function get_standard_source_text_for_attachment( int $attachment_id ): string {
+
+		if ( ! self::use_standard_source( $attachment_id ) ) {
+			//return '';
+		}
+
+		if ( self::standard_source_is( 'author_name' ) ) {
+			$author = get_post_field( 'post_author', $attachment_id );
+			$source = ! empty( $author ) ? get_the_author_meta( 'display_name', $author ) : '';
+		} else {
+			$source = self::get_standard_source_text();
+		}
+
+		/**
+		 * Filter the standard source text
+		 *
+		 * @param string $source standard source text.
+		 * @param int    $attachment_id attachment ID.
+		 */
+		return apply_filters( 'isc_standard_source_text_for_attachment', $source, $attachment_id );
+	}
+
+	/**
+	 * Verify the standard source option
+	 *
+	 * @param string $value value of the [standard_source] option.
+	 *
+	 * @return bool whether $value is identical to the standard source option or not.
+	 */
+	public static function standard_source_is( string $value ): bool {
+		$options = self::get_options();
+
+		if ( isset( $options['standard_source'] ) ) {
+			return $options['standard_source'] === $value;
+		}
+
+		/**
+		 * 2.0 moved the options to handle "own images" into "standard sources" and only offers a single choice for one of the options now
+		 * this section maps old to new settings
+		 */
+		if ( ! empty( $options['exclude_own_images'] ) ) {
+			return 'exclude' === $value;
+		} elseif ( ! empty( $options['use_authorname'] ) ) {
+			return 'author_name' === $value;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Get the standard source setting
+	 *
+	 * @return string
+	 */
+	public static function get_standard_source() {
+		$options = self::get_options();
+
+		// options since 2.0
+		if ( ! empty( $options['standard_source'] ) ) {
+			return $options['standard_source'];
+		}
+
+		/**
+		 * 2.0 moved the options to handle "own images" into "standard sources" and only offers a single choice for one of the options now
+		 * this section maps old to new settings
+		 */
+		if ( ! empty( $options['exclude_own_images'] ) ) {
+			return 'exclude';
+		} elseif ( ! empty( $options['use_authorname'] ) ) {
+			return 'author_name';
+		} elseif ( ! empty( $options['by_author_text'] ) ) {
+			return 'custom_text';
+		}
+
+		return '';
+	}
+
+	/**
+	 * Get the label of the standard source label
+	 *
+	 * @param string|null $value optional value, if missing, will use the stored value.
+	 *
+	 * @return string
+	 */
+	public static function get_standard_source_label( string $value = null ) {
+		$labels = array(
+			'exclude'     => __( 'Exclude from lists', 'image-source-control-isc' ),
+			'author_name' => __( 'Author name', 'image-source-control-isc' ),
+			'custom_text' => __( 'Custom text', 'image-source-control-isc' ),
+			'iptc'        => __( 'IPTC meta data', 'image-source-control-isc' ),
+		);
+
+		if ( ! $value ) {
+			$value = self::get_standard_source();
+		}
+
+		return $labels[ $value ] ?? '';
```

Comment:
**Suggestion:** Consider using a constant for the string 'custom_text' as it is used multiple times in the code. This will make the code more maintainable and less prone to errors. [maintainability]
```suggestion
  define('DEFAULT_STANDARD_SOURCE', 'custom_text');
  return $labels[ $value ] ?? DEFAULT_STANDARD_SOURCE;
```